lib_nbgl/src/nbgl_lns.c

```diff
@@ -0,0 +1,542 @@
+    screen_ctx.line2     = NULL;
+    screen_ctx.line2_len = 0;
+
+    if (screen_content.text == NULL) {
+        return false;
+    }
+
+    if (screen_content.icon != NULL) {
+        if (screen_content.centered) {
+            // Consider subtext is NULL
+            // Consider text fit in a single line
+            display_string(screen_content.text,
+                           screen_content.bold,
+                           screen_content.centered,
+                           TEXT_X_OFFSET,
+                           TEXT_Y_OFFSET_BELOW_ICON);
+            return true;
+        }
+        else {
+            // Override default line positioning to support icon on the left of the screen
+            lines_width = TEXT_WIDTH_BESIDE_ICON;
+            line1_x     = TEXT_X_OFFSET_BESIDE_ICON;
+            line1_y     = TEXT_Y_OFFSET_FIRST_LINE;
+        }
+    }
+
+    if (screen_content.subtext == NULL) {
+        uint8_t lines_nb
+            = nbgl_font_compute_nb_page(screen_content.text, lines_width, screen_content.bold);
+        if (lines_nb == 1) {
+            // Center the text on screen
+            line1_y = TEXT_Y_OFFSET_MIDDLE_LINE;
+        }
+        else if (lines_nb > 2) {
+            // No pagination supported in this mode
+            lines_nb = 2;
+        }
+
+        for (int line = 1; line <= lines_nb; line++) {
+            const char *start;
+            uint8_t     len;
+
+            nbgl_font_compute_paging(
+                screen_content.text, line, lines_width, screen_content.bold, &start, &len);
+            if (line == 1) {
+                line1     = screen_content.text;
+                line1_len = len;
+            }
+            else {
+                // line == 2
+                screen_ctx.line2      = start;
+                screen_ctx.line2_len  = len;
+                screen_ctx.line2_bold = screen_content.bold;
+            }
+        }
+    }
+    else {
+        // Consider text is single line
+        // subtext is forced to regular font
+        screen_ctx.line2_bold = false;
+
+        // Compute pagination parameters
+        if (screen_ctx.pagination_pages == 0) {
+            screen_ctx.pagination_pages
+                = nbgl_font_compute_nb_page(screen_content.subtext, lines_width, false);
+            if (screen_content.pos & BACKWARD_DIRECTION) {
+                screen_ctx.pagination_current_page = screen_ctx.pagination_pages;
+            }
+            else {
+                screen_ctx.pagination_current_page = 1;
+            }
+        }
+
+        if (screen_ctx.pagination_pages < 2) {
+            // No paging needed
+            line1 = screen_content.text;
+            if (screen_ctx.pagination_pages == 1) {
+                screen_ctx.line2 = screen_content.subtext;
+            }
+        }
+        else {
+            // Generate title with pagination
+            // Consider the resulting text fit in screen
+            char buffer[SCREEN_LINE_MAX_STRING_LEN];
+
+#if 1
```

Comment:
Which option will we keep ?

Reply:
I'm considering exposing `u8toa` somewhere apps could use it too. Though this is useful only for LNS, sparing quite a lot of flash if the app code doesn't need `snprintf`...
What do you think about it?

Reply:
Yes, could be good!